docs/docs/components/migrating/README.md

```diff
@@ -0,0 +1,190 @@
+# Migrating from Legacy Actions to Component Actions
+
+- [Overview](#overview)
+- [Key Changes](#key-changes)
+- [Getting Started](#getting-started)
+- [Migration Example](#migration-example)
+  * [Legacy Code Example](#legacy-code-example)
+  * [Converting into a Basic Component-Based Action](#converting-into-a-basic-component-based-action)
+  * [Advanced: Using Async Options](#advanced--using-async-options)
+  * [Publishing and Using Actions](#publishing-and-using-actions)
+
+## Overview
+This document is for developers who created legacy actions in [Pipedream's UI](https://pipedream.com/actions). The purpose is to help users migrate legacy actions to Pipedream's new [component model](/components). There is currently no deprecation timeline for legacy actions, but developers are encouraged to migrate to the new model. 
+
+## Key Changes
+
+**Capture user input via `props` instead of `params`** 
+
+The component model does not support `params`. You need to migrate `params` references to `props`. Unlike `params`, `props` must be explicitlly declared and defined prior to using them in code (in the old model, an input form was automatically generated when  `params` were used in code -- `params` were not explicitly declared).
+
+**Declare app `props` to use managed auth**
+
+The model for linking an app to legacy actions as well as the syntax for referencing credentials is different with Pipedream components. In the old model, apps were linked to steps in Pipedream's workflow builder UI, and credentials were referenced via the `auths` object.
+
+When using the component model, apps are defined as `props` and credentials are referenced as properties of the app. For example, to use managed auth for Github, the component `props` must contain a key (`gh` in the example below) with an app definition for the value (the app definition is an object):
+
+```javascript
+gh: {
+  type: "app",
+  app: "github"
+}
+```
+
+The component's `run()` method can then reference the credentials for Github via `this.gh.$auth.oauth_access_token`.
```

Comment:
It might be nice here to re-state that the auth details are stored in `this.gh.$auth`, and that different apps will expose their auth details in different properties, e.g. `api_key`. So it might be nice to tell the user that they can reference other example components for their app (if they exist), or run something like `console.log(Object.keys(this.gh.$auth))` to print the names of the auth properties.